pkg/didcomm/transport/http/inbound.go

```diff
@@ -19,29 +19,24 @@ import (
 
 var logger = log.New("aries-framework/transport")
 
-// provider contains dependencies for the HTTP Handler creation and is typically created by using aries.Context()
-type provider interface {
-	InboundMessageHandler() transport.InboundMessageHandler
-}
-
 // NewInboundHandler will create a new handler to enforce Did-Comm HTTP transport specs
 // then routes processing to the mandatory 'msgHandler' argument.
 //
 // Arguments:
 // * 'msgHandler' is the handler function that will be executed with the inbound request payload.
 //    Users of this library must manage the handling of all inbound payloads in this function.
-func NewInboundHandler(prov provider) (http.Handler, error) {
+func NewInboundHandler(prov transport.InboundProvider) (http.Handler, error) {
```

Comment:
Why this change? Note: if you need another provider in the future, you will need to break your API signature

Reply:
done